src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -58,13 +45,15 @@ ReadOnlyCollection<ParameterExpression> parameters
         /// <summary>
         /// Gets the parameters of the lambda expression.
         /// </summary>
-        public ReadOnlyCollection<ParameterExpression> Parameters => _parameters;
+        public ReadOnlyCollection<ParameterExpression> Parameters => GetOrMakeParameters();
 
         /// <summary>
         /// Gets the name of the lambda expression.
         /// </summary>
         /// <remarks>Used for debugging purposes.</remarks>
-        public string Name => _name;
+        public string Name => NameCore;
+
+        protected virtual string NameCore => null;
```

Comment:
Same as above. "Core" approach seems safe - just in case we have somebody calling this nonvirtually